Fix: ZFS_AC_KERNEL_SET_CACHED_ACL_USABLE check

ironMann opened the pull request:
Pass `ACL_TYPE_ACCESS` for type parameter of `get_cached_acl()` to avoid
removing dead code after BUG() in compile time. Tested on 3.2.0 kernel.

Introduced in 3779913

tuxoko commented:
@ironMann What about `forget_cached_acl`, you should add the type too for the same reason.

ironMann commented:
Fixed and rebased.

@tuxoko Adding the type parameter only once triggered the licence check. I added the same to `forget_cached_acl()` for completeness.
